Color-coding operations

distay0xGit opened the pull request:
This PR changes ArrayV to allow operations to be color-coded, with a legend included, if given operations it is able to color-code.

distay0xGit committed:
Add base color code functionality

distay0xGit committed:
Right-align color legend

Gaming32 commented:
Seems good, but I do have a few optimizations in mind... (such as not creating an array from a keyset on every frame)

distay0xGit committed:
Color-code some sorts

Gaming32 committed:
Merge branch 'main' into main

Gaming32 commented:
@distay0xGit Can you fix the checkstyle and build issues?

distay0xGit committed:
(hopefully) Make PR checkstyle-compliant

distay0xGit committed:
Merge branch 'main' of https://github.com/distay0xGit/ArrayV

Gaming32 committed:
Merge branch 'main' into pr/80

Gaming32 commented:
This is not a sorts PR. Can you please remove Explicium Sort?

distay0xGit committed:
Delete ExpliciumSort.java

Gaming32 committed:
Fix ignored checkstyle violation

distay0xGit committed:
Move legend slightly, properly clear colorcodes

distay0xGit committed:
Checkstyle violation

distay0xGit committed:
(rough draft) Subarray colorcoding

In desperate need of optimization, rough around the edges.

distay0xGit committed:
Uninitialized variables are not fun (ArrayVList.java)

distay0xGit committed:
Add ArrayVList colorcoding methods

And a method for writing colors between auxarrays. Thanks for suggesting it, Control.

distay0xGit committed:
Fix non-linked ScatterPlot colorcodes

distay0xGit committed:
Add colorcoding to more visual styles

distay0xGit committed:
Consistency

distay0xGit committed:
Stepping

distay0xGit committed:
Resolve conflicts attempt 1

distay0xGit committed:
Resolve conflicts attempt 2

GitHub is being stubborn

distay0xGit committed:
Resolve conflicts attempt 3

Seriously, GitHub?

distay0xGit committed:
Resolve conflicts attempt 4

This is getting stupid

distay0xGit committed:
"Fix" checkstyle

Gaming32 commented:
Git, *what are you doing*?

Gaming32 committed:
If Git is causing issues, that doesn't mean checkstyle should be ignored

We'll see if we can work out a better solution.

Gaming32 committed:
Merge branch 'main' into main

Gaming32 committed:
Fix oddly duplicated code

Git, what the heck?

Gaming32 committed:
Polymorphism

Gaming32 committed:
Some slight optimizations

distay0xGit committed:
Fix conflicts attempt 1

oops! bad coding practice

distay0xGit committed:
Fix conflicts attempt 2

AtomicInteger-related type changes I desperately want to avoid a conflict for and removing an interface from ArrayVlist

distay0xGit committed:
Fix conflicts attempt 3

_why doesn't it want me to change the packages imported for ArrayVList_

distay0xGit committed:
Fix conflicts attempt 4

the conflict checker is running off black magic and all the spells in it are specifically against pushing Groovy support into this PR

distay0xGit committed:
Fix conflicts attempt 5

I just hope I can revert this weird hack after this PR is brought up to speed with all the changes in main

rylid-umil commented:
Could the custom color coding be a toggleable option? like to switch between color-coded operations and normal highlighting?
